lib/tests/streamlit/metric_test.py

```diff
@@ -0,0 +1,182 @@
+# Copyright 2018-2021 Streamlit Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""metric unit tests."""
+from tests import testutil
+import streamlit as st
+
+from streamlit.errors import StreamlitAPIException
+from streamlit.proto.Metric_pb2 import Metric as MetricProto
+
+
+class MetricTest(testutil.DeltaGeneratorTestCase):
+    """Test ability to marshall metric protos and invalid input."""
+
+    def test_no_value(self):
+        st.metric("label_test", None)
+        c = self.get_delta_from_queue().new_element.metric
+        self.assertEqual(c.label, "label_test")
+        # This is an em dash. Not a regular "-"
+        self.assertEqual(c.body, "—")
+
+    def test_label_and_value(self):
+        """Test that metric can be called with label and value passed in."""
+        st.metric("label_test", "123")
+
+        c = self.get_delta_from_queue().new_element.metric
+        self.assertEqual(c.label, "label_test")
+        self.assertEqual(c.body, "123")
+        self.assertEqual(c.color, MetricProto.MetricColor.GRAY)
+        self.assertEqual(c.direction, MetricProto.MetricDirection.NONE)
+
+    def test_label_and_value_and_delta_and_delta_colors(self):
+        """Test that metric can be called with label and value passed in."""
```

Comment:
nit: the docstring should probably say "...label, value, delta, and delta colors..."